contributors/design-proposals/pod-preemption.md

```diff
@@ -0,0 +1,187 @@
+This implies that Best Effort pods are more likely to be evicted among a set of pods with the same priority. The reason is that any amount of resource usage by Best Effort pods translates into a very large percentage of usage over "requests", as Best Effort pods have zero requests for resources. So, while scheduler does not preempt Best Effort pods for releasing resources on a node, it is likely that these pods are evicted by the Kubelet after scheduler schedules a higher priority pod on the node.  
+Here is an example:
+
+1. Assume we have a node with 2GB of usable memory by pods.
+1. The node is running a burstable pod that uses 1GB of memory and 500MB memory request. It also runs a best effort pod that uses 1GB of memory (and 0 memory request). Both of the pods have priority of 100.
+1. A new pod with priority 200 arrives and asks for 2GB of memory.
+1. Scheduler knows that it has to preempt the burstable pod. From scheduler's point of view, the best effort pod needs no resources and its preemption will not release any resources.
+1. Scheduler preempts the burstable pod and schedules the high priority pending pod on the node.
+1. The high priority pod uses more than 1GB of memory. Kubelet detects the resource pressure and kills the best effort pod.
+
+So, best effort pods may be killed to make room for higher priority pods, although the scheduler does not preempt them directly.  
+Now, assume everything in the above example, but the best effort pod has priority 300. In this scenario, scheduler schedules the pending pod with priority 200 on the node, but it is evicted by the Kubelet, because the best effort pod has a higher priority. Given this scenario, scheduler should avoid the node and should try scheduling the pod on a different node if the pod is evicted by the Kubelet.
+
+## Race condition in multi-scheduler clusters
+
+Kubernetes allows a cluster to have more than one scheduler. This introduces a race condition where one scheduler (scheduler A) may perform preemption of one or more pods and another scheduler (scheduler B) schedules a different pod than the initial pending pod in the space opened after the preemption of pods and before the scheduler A has the chance to schedule the initial pending pod. In this case, scheduler A goes ahead and schedules the initial pending pod on the node thinking that the space is still available. However, the pod from A will be rejected by the kubelet admission process if there are not enough free resources on the node after the pod from B has been bound (or any other predicate that kubelet admission checks fails). This is not a major issue, as schedulers will try again to schedule the rejected pod.  
```

Comment:
Practically yes. What happens is that the status of the pod is set to failed and the corresponding controller creates a replacement pod which is getting scheduled later again.